mkreportindex.py

```diff
@@ -0,0 +1,111 @@
+import argparse
+import collections
+import os
+import base64
+import sys
+sys.path.append('/afs/cern.ch/user/d/daho/SAS-build/scripts')
+import common
+import checkformat
+
+def get_diff_map(file_name):
+    '''
+    generates the diffmap for a single file
+    '''
+    clang_format_output, num_violations = common.CheckFormattingRules(file_name)
+    diff_lines = common._ComparisonReport(file_name, clang_format_output)
+    if num_violations != 0:
+        return diff_lines
+    else:
+        return False
+
+def report_name(path, file_name):
+    '''
+    Generates a name for the report file based on the path and file name
+    '''
+    path_hash = base64.b64encode(path)
+    return "report_{path_hash}_{file_name}.html".format(path_hash=path_hash, file_name=file_name)
+
+
+def write_report_files(source_dir, report_dir):
+    '''
+    writes an html report file for each file in the source directory with coding standard violations
+    '''
+    checkformat.set_env()
+
+    error_dict = collections.OrderedDict()
+    total_errors = 0
+
+    for path, directories, files in os.walk(source_dir):
+        source_files = filter(common._IsSourceFile, files)
+        error_dict[path] = []
+        for file_name in source_files:
+            diff_map = get_diff_map(path + "/" + file_name)
+            if diff_map:
+                total_errors += 1
+                error_dict[path].append(file_name)
+                diff_map_name = report_name(path, file_name)
+                with open(os.path.join(report_dir, diff_map_name), "w") as fobj:
+                    fobj.write(diff_map)
+    return (error_dict, total_errors)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser("Formatter summary creator", formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('source_dir', type=str, help='path to the source directory')
+    args = parser.parse_args()
+    report_dir = "./report"
+
+    llvm = "/afs/cern.ch/sw/lcg/external/llvm/3.6/x86_64-slc6/"
+    os.environ["LD_LIBRARY_PATH"] = llvm+"lib64:"+os.environ["LD_LIBRARY_PATH"]
+    os.environ["PATH"] = llvm+"bin:"+os.environ["PATH"]
+    if not os.path.isfile(".clang-format"):
+        shutil.copy("/afs/cern.ch/exp/fcc/sw/0.6/SAS/0.1.4/x86_64-slc6-clang3.6-opt/config/FCCSW-clang-format.yaml", ".clang-format")
```

Comment:
you're already doing part of this in the `write_report_files` when calling `checkformat.set_env()`, so why not do it here directly?